src/main/java/appeng/util/item/UnsortedItemList.java

```diff
@@ -0,0 +1,176 @@
+ * Also no advanced features like storing craftable or requestable items is supported.
+ *
+ */
+public final class UnsortedItemList implements IItemList<IAEItemStack>
+{
+
+	/**
+	 * {@link Predicate} to filter all meaningful entries with {@link Iterators}
+	 */
+	private static final Predicate<IAEItemStack> MEANINGFUL_PREDICATE = new Predicate<IAEItemStack>(){
+
+		@Override
+		public boolean apply( IAEItemStack input )
+		{
+			return input.isMeaningful();
+		}
+	};
+
+	private final Map<IAEItemStack, IAEItemStack> records = new HashMap<IAEItemStack, IAEItemStack>();
+
+	@Override
+	public void add( IAEItemStack option )
+	{
+		if( option == null )
+		{
+			return;
+		}
+
+		final IAEItemStack st = this.records.get( option );
+
+		if( st != null )
+		{
+			st.add( option );
+			return;
+		}
+
+		final IAEItemStack opt = option.copy();
+
+		this.records.put( opt, opt );
+	}
+
+	@Override
+	public IAEItemStack findPrecise( IAEItemStack itemStack )
+	{
+		if( itemStack == null )
+		{
+			return null;
+		}
+
+		return this.records.get( itemStack );
+	}
+
+	@Override
+	public boolean isEmpty()
+	{
+		return this.records.isEmpty();
+	}
+
+	@Override
+	public int size()
+	{
+		return this.records.size();
+	}
+
+	@Override
+	public Iterator<IAEItemStack> iterator()
+	{
+		return Iterators.filter( this.records.values().iterator(), MEANINGFUL_PREDICATE );
+	}
+
+	@Override
+	public IAEItemStack getFirstItem()
+	{
+		for( IAEItemStack stackType : this )
+		{
+			return stackType;
+		}
+
+		return null;
+	}
+
+	@Override
+	public void resetStatus()
+	{
+		for( IAEItemStack i : this )
+		{
+			i.reset();
+		}
+	}
+
+	/**
+	 * Unsupported due to being a unsorted collection and thus only solvable in linear or worse time.
+	 *
+	 * @deprecated to indicate this method is unsupported.
+	 */
+	@Override
+	@Deprecated
+	public Collection<IAEItemStack> findFuzzy( IAEItemStack filter, FuzzyMode fuzzy )
+	{
+		throw new UnsupportedOperationException( "Unsupported on an unsorted collection" );
+	}
+
+	/**
+	 * Unsupported to avoid being used as anything but a plain collection to store real item stacks.
+	 *
+	 * @deprecated to indicate this method is unsupported.
+	 */
+	@Override
+	@Deprecated
+	public void addStorage( IAEItemStack option )
+	{
+		throw new UnsupportedOperationException( "Purely designed for item storage" );
+	}
+
+	/**
+	 * Unsupported to avoid being used as anything but a plain collection to store real item stacks.
+	 *
+	 * @deprecated to indicate this method is unsupported.
+	 */
+	@Override
+	@Deprecated
+	public void addCrafting( IAEItemStack option )
+	{
+		throw new UnsupportedOperationException( "Purely designed for item storage" );
+	}
+
+	/**
+	 * Unsupported to avoid being used as anything but a plain collection to store real item stacks.
+	 *
+	 * @deprecated to indicate this method is unsupported.
+	 */
+	@Override
+	@Deprecated
+	public void addRequestable( IAEItemStack option )
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)